Implement character length lt gt methods

akaryu0206 opened the pull request:
# New Features
- Implement `.lt?` method (#23)
- Implement `.lte?` method (#25)
- Implement `.gt?` method (#24)
- Implement `.gte?` method (#26)
- Add test for above methods

akaryu0206 committed:
[Add] Implement character length lt gt methods

- Implement `.lt?` method
- Implement `.lte?` method
- Implement `.gt?` method
- Implement `.gte?` method
- Add test for above methods

jagaapple commented on lib/string_foundation/length.rb:
```diff
@@ -14,4 +14,40 @@ def length?(length)
+  private
+
+  def self.allow_only_integer_argument(argument)
+    unless [Integer, Fixnum, Bignum].include?(argument.class)
```
I like the following.

```rb
return if [Integer, Fixnum, Bignum].include?(argument.class)

raise ...
```

jagaapple commented on lib/string_foundation/length.rb:
```diff
@@ -14,4 +14,40 @@ def length?(length)
+
+  private
+
+  def self.allow_only_integer_argument(argument)
```
Also change to a private instance method and use `instance_eval` method when using this.

jagaapple commented on lib/string_foundation/length.rb:
```diff
@@ -14,4 +14,40 @@ def length?(length)
+
+  private
+
+  def self.allow_only_integer_argument(argument)
```
Yes. Should write comments for all methods.

jagaapple commented on spec/string_foundation/length_spec.rb:
```diff
@@ -3,13 +3,13 @@
 # frozen_string_literal: true
 describe "[ Length Methods ]" do
+  let(:string) { "string" }
+  let(:expected_length) { 0 }
```
Good!

akaryu0206 committed:
[Clean] Refactor to reviewed codes

 
- Add comment to no comment method
- Add comment to private method
- Code beautify

Commit:
[Clean] Modify English for tests

- Modify English for tests

Commit:
[Add] Add documentation to a readme file

- Add documentation to a readme file for The Length Methods
